usr/share/rear/lib/global-functions.sh

```diff
@@ -387,17 +432,33 @@ output_path() {
 
 
 ### Mount URL $1 at mountpoint $2[, with options $3]
-mount_url() {
+function mount_url() {
     local url=$1
     local mountpoint=$2
     local defaultoptions="rw,noatime"
     local options=${3:-"$defaultoptions"}
+    local scheme
+
+    scheme=$( url_scheme $url )
```

Comment:
Only out of curiosity:
Is there a special reason why `scheme` is assigned here
in a separated command and not directly via
```
local scheme=$( url_scheme $url )
```
like at other code places `local scheme=$( url_scheme $BACKUP_URL )`

Reply:
The reason is explained here: https://github.com/koalaman/shellcheck/wiki/SC2155

I did not want to change all current code places where `local foo=$(bar)` is used, but at the same time I did not want to introduce more instances of this suboptiomal idiom, so it resulted in some inconsistency between the current and added code.

Reply:
Thank you for the explanation! I didn't know that.
So code like
```
local var=$(COMMAND) || Error "COMMAND failed"
```
won't work.
Fortunately is seems we don't have such code because
```
# find usr/share/rear -type f | xargs grep 'local .*||'
# find usr/share/rear -type f | xargs grep 'if .*local '
```
show no such code.

Reply:
yes, but it defeats the purpose of `set -e`. Compare
```bash
set -e
local foo=$(false)
```
and
```bash
set -e
local foo
foo=$(false)
```
(the second aborts, the first one does not).
I know that `set -e` probably won't work very well currently, so fixing this does not have a high priority.

Reply:
Thank you for the explanation how `local var=$(COMMAND)` doesn't work with `set -e`!
I didn't have `set -e` in mind.
In general modal behaviour calls for troubles (one always forgets some unusual mode).